src/abstractdataframe/iteration.jl

```diff
@@ -30,6 +30,10 @@ Base.iterate(::AbstractDataFrame) =
 Return a `DataFrameRows` that iterates a data frame row by row,
 with each row represented as a `DataFrameRow`.
 
+Because `DataFrameRow`s have an `eltype` of `Any`, use `copy(dfr::DataFrameRow)` to obtain 
+a named tuple, which supports iteration and property access like a `DataFrameRow`, 
+but also passes information of the `eltypes` of the columns of `df`. 
```

Comment:
this is a good idea but then the definition of:
```
Base.copy(r::DataFrameRow) = NamedTuple{Tuple(keys(r))}(values(r))
```
should be updated to reflect it (we should also pass `eltypes` of columns when constructing a `NamedTuple`).

Now it does not work as you describe.

When you add it please also add a test of this behavior.